File: LeetCode/Medium/0049. Group Anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        mp = {}

        for word in strs:
            ordered_word = "".join(sorted(word))

            if mp.get(ordered_word) is not None:
                mp.get(ordered_word).append(word)
            else:
                mp[ordered_word] = [word]
        
        res = []
        for word_list in mp.values():
            res.append(word_list)
        
        return res

File: LeetCode/Medium/0049. Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        vector<vector<string>> res;

        for(int i = 0; i < strs.size(); i++) {
            string s = strs[i];
            sort(s.begin(), s.end());
            cout << s << endl;
            
            mp[s].push_back(strs[i]);
        }
        
        for(auto it = mp.begin(); it != mp.end(); it++) {
            res.push_back(it->second);
        }
        return res;
    }
};
